src/core/algorithms/cfd/cfd_verifier/cfd_stats_calculator.cpp

```diff
@@ -0,0 +1,111 @@
+#include "cfd_stats_calculator.h"
+
+#include <ranges>
+
+#include "cfd/model/cfd_types.h"
+#include "cfd/util/cfd_output_util.h"
+
+namespace algos::cfd_verifier {
+void CFDStatsCalculator::CreateSupportMask() {
+    support_mask_.resize(relation_->GetNumRows(), true);
+
+    for (size_t i = 0; i < relation_->GetNumRows(); i++) {
+        cfd::Transaction const& row_values = relation_->GetRow(i);
+
+        support_mask_[i] = std::ranges::all_of(rule_.first, [&row_values](int val) {
+            return val <= 0 || std::ranges::find(row_values, val) != row_values.end();
+        });
+    }
+}
+
+void CFDStatsCalculator::MakeLhsToRowNums() {
+    for (size_t row_idx = 0; row_idx < relation_->GetNumRows(); ++row_idx) {
+        if (!support_mask_[row_idx]) {
+            continue;
+        }
+
+        cfd::Itemset lhs_values;
+        lhs_values.reserve(lhs_attrs_.size());
+        auto const& row_values = relation_->GetRow(row_idx);
+        for (int attr_idx : lhs_attrs_) {
+            lhs_values.push_back(row_values[attr_idx]);
+        }
```

Comment:
I think `insert` will be a bit more readable:
```suggestion
        lhs_values.insert(lhs_values.end(), row_values.begin(), row_values.end());
```
This also makes `reserve` unneeded.

Reply:
Thanks for the suggestion! However, this change copies all values from `row_values`, while the original code selects only those at indices in `lhs_attrs_`.